src/coreComponents/managers/TimeHistory/TimeHistoryCollection.hpp

```diff
@@ -102,8 +102,11 @@ class HistoryCollection : public TaskBase
       //  function pointer is actually assigned (an error would be thrown on the call attempt even so)
       GEOSX_ERROR_IF( m_bufferCalls[collectionIdx] == nullptr,
                       "History collection buffer retrieval function is unassigned, did you declare a related TimeHistoryOutput event?" );
-      // using GEOSX_ERROR_IF_EQ causes type issues since the values are used in iostreams
+      // using GEOSX_ERROR_IF_EQ caused type issues since the values are used in streams
+      // TODO : grab the metadata again, determine if the size has changed, update the buffer call if so...
       buffer_unit_type * buffer = m_bufferCalls[collectionIdx]();
+      DomainPartition & domain = dynamicCast< DomainPartition & >( *domain_group );
+      updateSetsIndices( domain );
       collect( domain, time_n, dt, collectionIdx, buffer );
```

Comment:
@wrtobin @rrsettgast  I had to add this `updateSetsIndices` here in order for the time history to work with the `faceElements`. Creating the set of indices earlier does not seem to work because the `faceElements` do not exist. I am not sure if doing this creates a performance issue. If so, we could probably only update the indices if they have changed.